index.md

```diff
@@ -0,0 +1,287 @@
+
+Your output format should play well with diffs (so ideally a line per thing-that-might-change, but apply common sense and non-default diffing tools if your output is e.g. naturally [tabular](https://github.com/paulfitz/daff) or a [bitmap](https://github.com/mapbox/pixelmatch)).
+
+Don't be afraid to make changes to the output format when you already have thousands of blessed tests - just make sure to run all tests before and after the output formatting changed without any changes anywhere but the testing infrastructure, and you can comfortably re-bless all the tests after a cursory glance at just a few.
+
+Your output format should be deterministic and shouldn't contain things that change often and nobody cares: for example, you will probably want to wrap each case with a "try..catch" and print stack traces into the output file (to enable a mix of regular and `assertThrows()` tests), but you might want to remove line numbers from the stack traces or every throwing case will need to be re-blessed every time you change anything in the code, and if your language includes paths in stack traces, you'll want to remove the part above the project root if you want different team members to be able to run the same test.
+
+If your output format can be readable to non-technical business stakeholders, that's a very nice bonus because it improves communication about requirements.
+
+If you write nontrivial test scaffolding that doesn't obviously get exercised well by your tests, test it separately. For example, you might want to dedicate some test cases just to the output formatting being correct, but if you add a new feature to your output formatting that is only triggered by one new case you just added and doesn't break any other test, it's ok to trust your test suite to catch mistakes in the new formatting code.
+
+## Further techniques
+
+There are many patterns that appear often when you approach writing tests this way, and they will be covered more extensively later.
+
+### VCR
+
+If you have an external dependency that exposes an API (e.g. REST or a DB), you might want to use the [VCR](https://github.com/vcr/vcr) pattern to be able to run your tests either in "record" (integration test) mode or in "playback" (unit test, mocked dependencies) mode.
+
+Don't get confused - you will have two kinds of recordings in your git repo: the requests and responses made by the test, as well as the blessed test output. Both are needed, they play different roles and should be stored separately.
+
+## A paradigm shift, not a hype cycle
+
+For a successful paradigm shift, first we must understand the existing paradigm and what it gets right, otherwise you just get a hype cycle[^1]. So here is a quick overview of unit testing methodology, and if any of this isn't familar, consider reading e.g. the testing chapters from [this](https://abseil.io/resources/swe-book/html/toc.html).
+
+### Why we write unit tests the way we do
```

Comment:
This section is not yet finished, right?  

Reading both this title and "How does this achieve our requirements?" list below, it's not obvious whether you're talking of how people presently write tests (which can only be an over-generalization but oh well), vs. how you're proposing to do it.

Same about "DAMP not DRY" — it's unclear at that point if you agree with it, or are gonna criticize it (as I suspect from "The DRY Testing Manifesto" :sweat_smile:)